### PR TITLE
Issue/6927 site creation details screen

### LINKED ISSUE #6927: Site-creation rework: Site details screen
Implement the 3rd step of the site-creation wizard, the site details screen.

| Step 3: Site Details |
| --- |
| screen shot 2017-11-30 at 23 13 46 |

### PR DESCRIPTION
**Pull request by hypest**
Fixes #6927 

~Needs #7028 to be merged first!~ Done now.

Introduces the Site Details screen, asking the user for the site title (mandatory) and the site tagline (optional).

To test:
1. With the app logged in, head over to MySite and tap on the "Switch Site" button
2. Tap on the "+" action on the toolbar
3. Select "Creation WordPress.com site" option from the popup
4. Tap on any of the category cards
5. Tap on any of the themes
6. Notice the Site Details screen and compare it with the design [on the ticket](https://github.com/wordpress-mobile/WordPress-Android/issues/6927)
7. Try out the input boxes and the "Next" button
8. Upon successful completion of the form, a Toast pops up with the inputted values for the site title and tagline

### CODE REVIEW

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java [hunk 1]
```diff
@@ -770,6 +770,8 @@ public static String getEventNameForStat(AnalyticsTracker.Stat stat) {
                 return "site_creation_category_viewed";
             case SITE_CREATION_THEME_VIEWED:
                 return "site_creation_theme_viewed";
+            case SITE_CREATION_DETAILS_VIEWED:
+                return "site_creation_details_viewed";
```
Comment (theck13) [inline libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: Let's make sure this statistic is added to the tracking documentation.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationBaseFormFragment.java [hunk 2]
```diff
@@ -15,16 +15,24 @@
...
     protected abstract void setupContent(ViewGroup rootView);
 
+    protected void setupBottomButtons(Button primaryButton) {};
+    protected Button getPrimaryButton() {
+        return mPrimaryButton;
+    }
+
     protected abstract void onHelp();
 
     @Override
```
[the remark this answers is not in the document]
Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationBaseFormFragment.java]: Hah, weird that has happened! Fixed with ec06994.

File: WordPress/src/main/res/drawable/ic_gridicons_aside.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M14,20l6,-6L20,6c0,-1.105 -0.895,-2 -2,-2L6,4c-1.105,0 -2,0.895 -2,2v12c0,1.105 0.895,2 2,2h8zM6,6h12v6h-4c-1.105,0 -2,0.895 -2,2v4L6,18L6,6zM16,10L8,10L8,8h8v2z"/>
+</vector>
```
[the remark this answers is not in the document]
Comment (hypest) [reply WordPress/src/main/res/drawable/ic_gridicons_aside.xml]: Makes sense, done with 13d9e44.

File: libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java (hunk 1 above)
Comment (hypest) [reply libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTrackerNosara.java]: Done!

**Comment by hypest** [comment]
Thanks for the review @theck13 !

> ...it overlaps the tagline description. 

Oops! Should be opaque and the main content above the button. Fixed with f7c2bdb , 61b02d2.

> I noticed the Next button behaves a little differently than the login flow. The button is disabled until text is entered rather than being always enabled ...

True, this was a design decision. @folletto can confirm.

Ready for another pass, thanks!

**Comment by folletto** [comment]
Yep would be nice to align, but it's ok in this context. :)

**Pull request merged**